pkg/util/arm/marshal.go

```diff
@@ -42,6 +67,17 @@ func (r *Resource) MarshalJSON() ([]byte, error) {
 		if r.DependsOn != nil {
 			dataMap["dependsOn"] = r.DependsOn
 		}
+		resourceType := reflect.TypeOf(r.Resource)
+		elemType := resourceType
+		if resourceType.Kind() == reflect.Ptr {
+			elemType = resourceType.Elem()
+		}
+		if (armResourceTypes[resourceType] || armResourceTypes[elemType]) && r.Type != "" {
```

Comment:
I did some debugging to understand the logic here a bit better and whenever this block is entered `dataMap["type"]` is nil and vice versa. (whenever `dataMap["type"]==nil`, this block was entered.).

Would it make sense to replace this cross checking against `armResourceTypes` with a simple nil-check?
```
if dataMap["type"] == nil && r.Type != ""
```